zerver/views/reminders.py

```diff
@@ -0,0 +1,36 @@
+from django.http import HttpRequest, HttpResponse
+from django.utils.timezone import now as timezone_now
+from django.utils.translation import gettext as _
+from pydantic import Json
+
+from zerver.actions.reminders import schedule_reminder_for_message
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.request import RequestNotes
+from zerver.lib.response import json_success
+from zerver.lib.timestamp import timestamp_to_datetime
+from zerver.lib.typed_endpoint import typed_endpoint
+from zerver.models import UserProfile
+
+
+@typed_endpoint
+def create_reminders_message_backend(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    message_id: Json[int],
+    scheduled_delivery_timestamp: Json[int],
+) -> HttpResponse:
+    deliver_at = timestamp_to_datetime(scheduled_delivery_timestamp)
+    if deliver_at <= timezone_now():
+        raise JsonableError(_("Scheduled delivery time for reminder must be in the future."))
```

Comment:
I think probably we want to share the string `Scheduled delivery time must be in the future.` with the scheduled message code path -- maybe best would be a new `expections.py` class for that, but if not just use the same string.

Reply:
Added a prep commit to do that.